examples/vmi-sriov.yaml

```diff
@@ -1,3 +1,14 @@
+apiVersion: sriovnetwork.openshift.io/v1
```

Comment:
i dont think it should be part of the vmi yaml file, it can be on a separate file or in some readme imo

Reply:
> ii am not sure it should be part of the vmi yaml file, it can be on a separate file or in some readme,
> i do understand its more comfortable, just usually vmi examples have just the vmi, unless we decide its ok to have the whole package in it.
> 

> Another reason is that if you delete the vm, you wont be able to reuse the file again out of the box since the network is already created.

You can still use this file again even if the network is there, you will just get a message saying `sriovnetwork/xxxx not changed` something like that

Reply:
This is not good since the example is generated from https://github.com/kubevirt/kubevirt/blob/385a9d49b6d840a9fe5803e3a5083428b0eaaebe/tools/vms-generator/utils/utils.go#L449 . You will have to patch that (you'll see that `make generate` will fail).

Other than that, good discussion on if the file should have all the dependencies; I think this warrants an email to the kubevirt-dev list, since it has impacts outside of networking. 

My personal opinion: given how much simpler it is to the user to find everything needed to have a functional example, I think the example files should be self-contained.

Reply:
>  I think this warrants an email to the kubevirt-dev list

I would try first on the community meeting and per the feedback there go with the email or just directly on this PR.

Reply:
maybe a folder where the yamls are a compound or a file name that hints about it being a compound is a way to go ?

Reply:
@maiqueb I had no idea this is how exampls are generated thanks for the heads-up,
I guess any change should be done to this code and not directly to the example files.
This yaml file is self-contained even though it includes SriovNetwork there is no problem with creating and deleting it using `kubectl apply -f` if that what you mean.

I will raise this question if a yaml should contain all dependent objects or not, and if not then where
 on the next community meeting

Reply:
> @maiqueb I had no idea this is how exampls are generated thanks for the heads-up,

Now you know; you're welcome.

> I guess any change should be done to this code and not directly to the example files.

Yes. 

> This yaml file with this PR changes is self-contained even though it includes SriovNetwork there is no problem with creating and deleting it using `kubectl apply -f` if that what you mean.

I never said there would be a problem with this file - all I said is if I have to choose between a file with the full example or multiple files with pieces of the example, I would choose a single self-contained file.

> 
> I will raise this question if a yaml should contain all dependent objects or not, and if not then where
> on the next community meeting

Thanks, let's see what comes out of it.

Reply:
Now that I realized that example files generated by vms-generator code, after I reviewed it 
and the fact that we still have dont have SriovNetwork library.
I think its better to create SriovNetwork yaml file on its own.